src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
 import PropTypes from 'prop-types';
 
-const ChatEntry = (props) => {
+const ChatEntry = ( { sender, body, timeStamp, likeMessage, unlikeMessage } ) => {
+  const [hearted, setHearted] = useState(false);
+
+  const toggleHeart = () => {
+    setHearted(hearted => !hearted);
+    hearted ? unlikeMessage(): likeMessage();
+  };
```

Comment:
Following my suggestion about passing down an event handler that can be used to change a message's 'liked' status, you would use that event handler in toggleHeart here:

```javascript
  const toggleHeart = () => {
    props.updateEntryData(props.id);
  };